io.go

```diff
@@ -44,6 +47,42 @@ func (cr *CborReader) SetReader(r io.Reader) {
 	cr.r = GetPeeker(r)
 }
 
+func (cr *CborReader) ReadCid(scratchBuf []byte) (cid.Cid, error) {
```

Comment:
Were you planning on using this somewhere?

Reply:
outside of package, yeah. Its really a separate thing from the rest of this PR, happy to remove for now and bring back in later

Reply:
It's just kind of awkward given that:

1. Everything else uses the internal scratch buffer.
2. It will fail if the scratch buffer isn't large enough.

If we need a CID scratch buffer, I'd consider just increasing the size of the internal scratch buffer to fit CIDs.

Reply:
alright, ill just remove this for now. can bring it back later